src/test/java/org/mule/tools/api/packager/ValidatorTest.java

```diff
@@ -0,0 +1,175 @@
+import org.junit.rules.TemporaryFolder;
+import org.mockito.Mockito;
+
+public class ValidatorTest {
+
+  public static final String MULE_DOMAIN = "mule-domain";
+  public static final String MULE_APPLICATION = "mule-application";
+  public static final String MULE_POLICY = "mule-policy";
+  @Rule
+  public TemporaryFolder projectBaseFolder = new TemporaryFolder();
+
+  private Validator validator;
+
+  @Before
+  public void before() throws IOException, MojoExecutionException {
+    validator = new Validator(projectBaseFolder.getRoot().toPath());
+  }
+
+  @Test
+  public void isMuleApplicationPackagingTypeValid() throws ValidationException {
+    Boolean valid = validator.isPackagingTypeValid(MULE_APPLICATION);
+    assertThat("Packaging type should be valid", valid, is(true));
+  }
+
+  @Test
+  public void isMuleDomainPackagingTypeValid() throws ValidationException {
+    Boolean valid = validator.isPackagingTypeValid(MULE_DOMAIN);
+    assertThat("Packaging type should be valid", valid, is(true));
+  }
+
+  @Test
+  public void isMulePolicyPackagingTypeValid() throws ValidationException {
+    Boolean valid = validator.isPackagingTypeValid(MULE_POLICY);
+    assertThat("Packaging type should be valid", valid, is(true));
+  }
+
+  @Test(expected = ValidationException.class)
+  public void isPackagingTypeValid() throws ValidationException {
+    Boolean valid = validator.isPackagingTypeValid("no-valid-packagin");
+    assertThat("Packaging type should be valid", valid, is(true));
+  }
+
+  @Test(expected = ValidationException.class)
+  public void projectStructureValidMuleApplicationInvalid() throws ValidationException {
+    Path mainSrcFolder = projectBaseFolder.getRoot().toPath().resolve(SRC.value()).resolve(MAIN.value());
+    mainSrcFolder.toFile().mkdirs();
+
+    validator.isProjectStructureValid(MULE_APPLICATION);
+  }
+
+  @Test
+  public void isProjectStructureValidMuleApplication() throws ValidationException {
+    Path muleMainSrcFolder =
+        projectBaseFolder.getRoot().toPath().resolve(SRC.value()).resolve(MAIN.value()).resolve(MULE.value());
+    muleMainSrcFolder.toFile().mkdirs();
+
+    Boolean valid = validator.isProjectStructureValid(MULE_APPLICATION);
+    assertThat("Project structure should be valid", valid, is(true));
+  }
+
+
+  @Test(expected = ValidationException.class)
+  public void isProjectStructureValidMulePolicyInvalid() throws ValidationException {
+    Path mainSrcFolder = projectBaseFolder.getRoot().toPath().resolve(SRC.value()).resolve(MAIN.value());
+    mainSrcFolder.toFile().mkdirs();
+
+    validator.isProjectStructureValid(MULE_POLICY);
+  }
+
+  @Test
+  public void isProjectStructureValidMulePolicy() throws ValidationException {
+    Path muleMainSrcFolder =
+        projectBaseFolder.getRoot().toPath().resolve(SRC.value()).resolve(MAIN.value()).resolve(POLICY.value());
+    muleMainSrcFolder.toFile().mkdirs();
+
+    Boolean valid = validator.isProjectStructureValid(MULE_POLICY);
```

Comment:
Would it be possible to check the case where the project that is being validated has a valid structure of another packaging type (for instance MULE_APPLICATION)?